extensions/guacamole-auth-ldap/src/main/java/org/apache/guacamole/auth/ldap/connection/ConnectionService.java

```diff
@@ -59,6 +62,42 @@
      * Logger for this class.
      */
     private static final Logger logger = LoggerFactory.getLogger(ConnectionService.class);
+    
+    /**
+     * The name of the LDAP attribute that stores connection configuration
+     * parameters for Guacamole.
+     */
+    public static final String LDAP_ATTRIBUTE_PARAMETER = "guacConfigParameter";
+    
+    /**
+     * The name of the LDAP attribute that stores the protocol for a Guacamole
+     * connection.
+     */
+    public static final String LDAP_ATTRIBUTE_PROTOCOL = "guacConfigProtocol";
+    
+    /**
+     * The name of the LDAP attribute that stores proxy configuration parameters
+     * for Guacamole.
+     */
+    public static final String LDAP_ATTRIBUTE_PROXY = "guacConfigProxy";
+    
+    /**
+     * The proxy parameter name that configures the hostname or IP address that
+     * will be used to connect to guacd.
+     */
+    public static final String PROXY_PARAMETER_HOSTNAME = "proxy-hostname";
+    
+    /**
+     * The proxy parameter name that configures the TCP port that will be used
+     * to connect to guacd.
+     */
+    public static final String PROXY_PARAMETER_PORT = "proxy-port";
+    
+    /**
+     * The proxy parameter name that configures the encryption method that will
+     * be used when connected to guacd.
+     */
+    public static final String PROXY_PARAMETER_ENCRYPTION = "proxy-encryption";
```

Comment:
I think these and the related parameter parsing should be gone in favor of the new `guacConfigProxy` attribute.

Reply:
I'm not sure I understand? These were added as part of the `guacConfigProxy` attribute - what should I be replacing them with?

Reply:
Oh, I thought these were holdovers from before `guacConfigProxy` when `guacConfigParameter` was being used. I misread the parsing leveraging these constants as that connection parameter parsing still being there, with `guacConfigProxy` being a direct retrieval elsewhere.

Reply:
Okay - if you've another suggestion for parsing out guacConfigProxy, I'm certainly not opposed to a better way, but this is just the way I had gone about implementing it - with constants for each of the possible parameter names for `guacConfigProxy`.

Reply:
Rather than parsing the content of `guacConfigProxy`, would it make more sense to do away with that and instead define `guacConfigProxyHostname`, `guacConfigProxyPort`, and `guacConfigProxyEncryption`?

`guacConfigParameter` has to be special since it defines connection parameters, and those must be arbitrary name/value pairs, but the parameters for connecting to guacd are not arbitrary name/value pairs.

Reply:
Yeah, that makes sense. I'll re-factor it for that.